File: javascript/grade-school/grade-school.js
export class GradeSchool {
  constructor() {
    this._roster = {};
  }

  roster = () => this.deepClone(this._roster);

  add = (name, grade) => {
    const students = this._roster[grade] || [];
    for (const [grade, students] of Object.entries(this._roster))
      if (students.includes(name)) this._roster[grade] = students.filter(student => student !== name);
    this._roster[grade] = [...students, name].sort();
  };

  grade = gr => this.deepClone(this._roster[gr ] || []);

  deepClone = obj => JSON.parse(JSON.stringify(obj));
}

File: julia/grade-school/grade-school.jl
new_school() = []

function add!(students, school)
      map(student ->
          if student.name in roster(school)
              return false
          else
            push!(school, student)
            return true
          end, students)
end

grade(num, school) = map(x -> x.name, school |> filter(x -> x.grade == num)) |> sort

roster(school) = map(x -> x.name, sort(school; by = x -> (x.grade, x.name)))
